src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,386 @@
+    std::vector<double> in_frac_ = { 0.80, 0.15, 0.05 };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    std::vector<Material::Ptr> mat;
+    mat.push_back(Material::CreateUntracked(in_cap[0], c_natu()));
+    mat.push_back(Material::CreateUntracked(in_cap[1], c_pustream()));
+    mat.push_back(Material::CreateUntracked(in_cap[2], c_uox()));
+    
+    SetInputInv(mat);
+    mf_facility_->Tick();
+    
+    CompMap v_0 = c_natu()->mass();
+    Normalize(&v_0, in_frac[0]);
+    CompMap v_1 = c_pustream()->mass();
+    Normalize(&v_1, in_frac[1]);
+    CompMap v_2 = c_uox()->mass();
+    Normalize(&v_2, in_frac[2]);
+    CompMap v = v_0;
+    v = Add( v, v_1);
+    v = Add( v, v_2);
+    
+    InvBuffer* buffer = GetOutPutBuffer();
+    Material::Ptr final_mat = cyclus::ResCast<Material>(buffer->PopBack());
+    cyclus::CompMap final_comp = final_mat->comp()->mass();
+    
+    Normalize(&v,1);
+    Normalize(&final_comp,1);
+    
+    cyclus::CompMap sum_v = Add(v, final_comp);
+    
+    CompMap::iterator it;
+    for(it = sum_v.begin(); it != sum_v.end(); it ++){
+      EXPECT_DOUBLE_EQ(final_comp[it->first],v[it->first]) << "Unexpected difference on nuclide " << it->first << ".";
+    }
+  }
+
+  TEST_F(MixingFabTest, Throughput) {
+    
+    std::vector<std::string> in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_frac_ = { 0.80, 0.15, 0.05 };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(0.5);
+    
+    std::vector<Material::Ptr> mat;
+    mat.push_back(Material::CreateUntracked(in_cap[0], c_natu()));
+    mat.push_back(Material::CreateUntracked(in_cap[1], c_pustream()));
+    mat.push_back(Material::CreateUntracked(in_cap[2], c_uox()));
+    SetInputInv(mat);
+    
+    mf_facility_->Tick();
+    
+    std::vector<double> cap;
+    for(int i = 0; i < in_com.size(); i++ ){
+      cap.push_back(in_cap[i]-0.5*in_frac[i]);
+    }
+    
+    std::map<std::string, InvBuffer> streambuf = GetStreamBuffer();
+    
+    for (int i = 0; i < in_com.size(); i++) {
+      std::string buf_com = in_com[i];
+      double buf_size = in_cap[i];
+      double buf_ratio = in_frac[i];
+      double buf_inv = streambuf[buf_com].quantity();
+      
+      EXPECT_EQ( buf_size - 0.5*buf_ratio, buf_inv) << " one (or more) input buffer inventory is not equal to the buffer size times the mixing ratio.";
+    }
+    
+    
+    // output buffer size should be equal to the throuput size
+    EXPECT_EQ(throughput, GetOutPutBuffer()->quantity()) << " output buffer inventory is not equal to the buffer size.";
```

Comment:
Maybe this should be a separate test? - you can decide.  And your message is inconsistent with the code - maybe should say ``mixing was not correctly constrained by throughput``?